src/timing.h

```diff
@@ -360,16 +370,17 @@ struct jl_timing_block_cpp_t {
     jl_timing_block_cpp_t& operator=(const jl_timing_block_cpp_t &) = delete;
     jl_timing_block_cpp_t& operator=(const jl_timing_block_cpp_t &&) = delete;
 };
-#define JL_TIMING(subsystem, event) jl_timing_block_cpp_t __timing_block(JL_TIMING_ ## subsystem, JL_TIMING_EVENT_ ## event); \
-    _TRACY_CTOR(__timing_block.block.tracy_ctx, #event, (jl_timing_enable_mask >> (JL_TIMING_ ## subsystem)) & 1)
+#define JL_COND_TIMING(enabled, subsystem, event) \
+    jl_timing_block_cpp_t COND_CAT(__timing_block, __LINE__)(enabled, JL_TIMING_ ## subsystem, JL_TIMING_EVENT_ ## event, #event);
 #else
-#define JL_TIMING(subsystem, event) \
+#define JL_COND_TIMING(enabled, subsystem, event) \
     __attribute__((cleanup(_jl_timing_block_destroy))) \
-    jl_timing_block_t __timing_block; \
-    _jl_timing_block_ctor(&__timing_block, JL_TIMING_ ## subsystem, JL_TIMING_EVENT_ ## event); \
-    _TRACY_CTOR(__timing_block.tracy_ctx, #event, (jl_timing_enable_mask >> (JL_TIMING_ ## subsystem)) & 1)
```

Comment:
This constructor is intentionally placed inline here because it's a macro that uses `__func__`/`__LINE__`/etc.

It can be lifted it into a function, but you'd have to expand the `TracyCZoneN` macro by hand and make sure the function call it wraps gets the right arguments